frontend/app/services/wsaddress-service.server.ts

```diff
@@ -75,7 +84,7 @@ async function createWSAddressService() {
       province: correctionResults['nc:ProvinceCode'],
       postalCode: correctionResults['nc:AddressPostalCode'],
       country: correctionResults['nc:CountryCode'],
-    };
+    } as CorrectAddressResponse;
```

Comment:
It's not needed since `correctAddress` is implicitly returning the type.

Reply:
Oh, I see what you mean now... let's investigate.

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/114004123/6f836dd7-af4f-4d63-a084-0dfbaee18b77)

Reply:
I've left the `interface` in for now but I am open to removing it and explicitly setting the fields as `undefined` in the unit test - I have no strong opinion on this.

Reply:
Today I learned that TypeScript typechecking treats optional `?:` differently from the undefined type. To resolve this issue, what was done is rather than declaring an interface, either the missing props should be added in the tests (using a defaultMock object and spreading, which is my preference), or the returned type of the function should be explicitly typed.

Preferred Option 1:

```typescript
// before describe('_gcweb-app.personal-information.home-address.edit', () => {
const baseCorrectAddressMockResolvedValue = {
  address: undefined,
  city: undefined,
  province: undefined,
  postalCode: undefined,
  country: undefined,
};

// when mocking
vi.mocked(wsAddressService.correctAddress).mockResolvedValue({
  ...baseCorrectAddressMockResolvedValue,
  status: 'Valid',
});
```

Option 2:

```typescript
: Promise<{
    status: string;
    address?: string;
    city?: string;
    province?: string;
    postalCode?: string;
    country?: string;
  }
```

The reason for preferring the first option is because it allows for code adjustments considering testing implementation.

Reply:
> the missing props should be added in the tests

Can you just use a partial mock?

``` typescript
vi.mocked(wsAddressService.correctAddress, { partial: true }).mockReturnValue({ ... })
```

Reply:
Thanks for suggestions.

Removed `CorrectAddressResponse` uses [here](https://github.com/DTS-STN/canadian-dental-care-plan/compare/3b8fa830bade1a4323acf74e1c574a7942d03d2c..0fb03103edb53651317d5b7096d9360c3bfb87d7#diff-a330c9ede4049228d6825394285f16e4e4737fb401eb33bdfaa0021111a6bb85).

Using a partial mock [here](https://github.com/DTS-STN/canadian-dental-care-plan/compare/3b8fa830bade1a4323acf74e1c574a7942d03d2c..0fb03103edb53651317d5b7096d9360c3bfb87d7#diff-a72bc829ebe90af4d2b71eaa29c4482e8a870cfa4fbd9ffbb1760d2a4154f881).